src/backend/distributed/executor/multi_utility.c

```diff
@@ -233,7 +232,7 @@ multi_ProcessUtility(Node *parsetree,
 	 * DDL commands are propagated to workers only if EnableDDLPropagation is
 	 * set to true and the current node is the schema node
 	 */
-	if (propagateChanges)
+	if (EnableDDLPropagation)
```

Comment:
Yes, even if the table is not the schema node, we have to go through the DDL processing functions and error out only with the DDL on distributed tables.